### PR TITLE
Jazzy patch release 2 versions 2024-09-19

### PR DESCRIPTION
**Pull request by marcoag**
First Jazzy patch release :tada: 

CI:
* Linux [Build Status](https://ci.ros2.org/job/ci_linux/21859/)
* Linux-aarch64 [Build Status](https://ci.ros2.org/job/ci_linux-aarch64/16129/)
* Windows [Build Status](https://ci.ros2.org/job/ci_windows/22570/)
* RHEL [Build Status](https://ci.ros2.org/job/ci_linux-rhel/1365/)

Packaging:
* Linux [Build Status](https://ci.ros2.org/job/ci_packaging_linux/562/)
* Linux-aarch64 [Build Status](https://ci.ros2.org/job/ci_packaging_linux-aarch64/109/)
* Windows [Build Status](https://ci.ros2.org/job/ci_packaging_windows/233/)
* RHEL [Build Status](https://ci.ros2.org/job/ci_packaging_linux-rhel/67/)

### CODE REVIEW

**Comment by marcoag** [comment]
The RHEL jobs and linux packaging failed with `No space left on device`, retrying:
* RHEL [Build Status](https://ci.ros2.org/job/ci_linux-rhel/1368/)

Packaging:
* Linux [Build Status](https://ci.ros2.org/job/ci_packaging_linux/563/)
* RHEL [Build Status](https://ci.ros2.org/job/ci_packaging_linux-rhel/69/)

**Comment by marcoag** [comment]
Last min update on rclcpp: https://github.com/ros2/rclcpp/pull/2628. Marking this as a draft so it doesn't get merge, will update once packages rebuild and versions are updated here.

**Comment by marcoag** [comment]
Reruning CI and packages jobs after the [rcl and rclcpp update](https://github.com/ros2/ros2/pull/1605/commits/8b5ca119f0edbc9a5c50252623d7403bc273a44a):

CI:
* Linux [Build Status](https://ci.ros2.org/job/ci_linux/21863/)
* Linux-aarch64 [Build Status](https://ci.ros2.org/job/ci_linux-aarch64/16133/)
* Windows [Build Status](https://ci.ros2.org/job/ci_windows/22574/)
* RHEL [Build Status](https://ci.ros2.org/job/ci_linux-rhel/1370/)

Packaging:
* Linux [Build Status](https://ci.ros2.org/job/ci_packaging_linux/564/)
* Linux-aarch64 [Build Status](https://ci.ros2.org/job/ci_packaging_linux-aarch64/110/)
* Windows [Build Status](https://ci.ros2.org/job/ci_packaging_windows/234/)
* RHEL [Build Status](https://ci.ros2.org/job/ci_packaging_linux-rhel/70/)

**Comment by marcoag** [comment]
Windows seemed to have some issue reaching chocolatey, rerunning:

* Windows [Build Status](https://ci.ros2.org/job/ci_windows/22575/)

File: ros2.repos [hunk 1]
```diff
@@ -282,19 +282,19 @@ repositories:
...
+    version: 7.1.2
   ros2/rcpputils:
     type: git
     url: https://github.com/ros2/rcpputils.git
     version: 2.11.0
   ros2/rcutils:
     type: git
     url: https://github.com/ros2/rcutils.git
-    version: 6.7.1
+    version: 6.7.2
```
Comment (clalancette) [inline ros2.repos:297]: The corresponding rosdistro PR hasn't been merged yet: https://github.com/ros/rosdistro/pull/42886 .  So we either need to merge that, or put this back to 6.7.1 and wait for that update until the next patch release.

**Review by clalancette — CHANGES_REQUESTED** [review]
Besides what is inline, the CI jobs are very yellow/red.  Is that expected?

File: ros2.repos (hunk 1 above)
Comment (marcoag) [reply ros2.repos:297]: Thanks for the catch, merged the PR, waiting for rebuild.

**Comment by marcoag** [comment]
As I mentioned the red was windows not being able to reach chocolatey packages, it seems due to some network issue. 

The linux yellow ones is a `cmake_minimum_required` warning which were there on the previous [patch](https://github.com/ros2/ros2/pull/1575).

RHEL is now throwing a couple of `no CONNEXTDDS_DIR nor NDDSHOME specified` as a warning, not sure how much we care about this but it seems tests are still passing without it.

**Comment by clalancette** [comment]
> RHEL is now throwing a couple of `no CONNEXTDDS_DIR nor NDDSHOME specified` as a warning, not sure how much we care about this but it seems tests are still passing without it.

Ah, that's because we stopped installing Connext for RHEL.  We should backport https://github.com/ros2/rmw_connextdds/pull/158, but it is not critical.

> I assume even though I just updated `rcutils` in the `rosdistro` file we don't have to rerun ci and packaging jobs since they were run with the `ros2.repos` from this PR, which includes `rcutils 6.7.2`.

Yes, agreed.

**Review by clalancette — APPROVED** [review]

**Pull request merged**